### PR TITLE
HW40 KaterinaP

### PR DESCRIPTION
**Pull request by KaterinaPogorelova**
login, password are the same

### CODE REVIEW

File: app-hw-37/src/Auth/Auth.tsx [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-					<label className="auth__label" htmlFor='login'>Login</label>
-					<input className="auth__input" type="text" id="login" placeholder='Your Login' value={this.state.login} onChange={(event) => this.setState({ login: event.target.value })} />
-					<label className="auth__label" htmlFor='password'>Password</label>
-					<input className="auth__input" type="password" id="password" placeholder='Your Password' value={this.state.pass} onChange={(event) => this.setState({ pass: event.target.value })} />
-					<button className="auth__button" type="submit" value="Submit" onClick={this.handleAuth}>Submit</button>
-				</form>
-			</>)
+
+export const Auth = (props: Props) => {
+	const [account, setAccount] = useState({ login: '', password: '' })
```
Comment (jathpr) [inline app-hw-37/src/Auth/Auth.tsx:10]: you need two states here
for login and for pass

File: app-hw-37/src/CardsList/Card.tsx [hunk 2]
```diff
@@ -15,27 +15,22 @@ type Cardprops = {
...
-				<div className="post__text">
-					<p className="post__date">{this.state.cardstate.date}</p>
-					<h3 className="post__title">{this.state.cardstate.title}</h3>
-					<p className='post__desc'>{this.state.cardstate.description}</p>
-					<p className='post__author'>Author: {this.state.cardstate.author} {this.state.cardstate.text}</p>
-				</div>
-			</button>
-		)
+
+export const Card = (props: Cardprops) => {
```
Comment (jathpr) [inline app-hw-37/src/CardsList/Card.tsx:19]: I recommend destructurization here
{cardinfo, showFullScreen}

File: app-hw-37/src/FullScreenCard/FullScreenCard.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	description: string,
+	author: number
+}
+
+type Props = {
+	id: number,
+	goBack: () => void
+}
+export const FullScreenCard = (props: Props) => {
+	const [post, setPost] = useState<Post>({
```
Comment (jathpr) [inline app-hw-37/src/FullScreenCard/FullScreenCard.tsx:21]: prefer undefined then empty object

File: app-hw-37/src/Main/Main.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	lesson_num: number,
+	title: string,
+	description: string,
+	author: number
+}
+
+
+export const Main = () => {
+	const [page, setPage] = useState('List')
+	const [fullScreenPostId, setFullScreenPostId] = useState(0)
```
Comment (jathpr) [inline app-hw-37/src/Main/Main.tsx:20]: prefer null or undefined then 0

**Review by jathpr — APPROVED** [review]
great work

**Comment by jathpr** [comment]
can be merged

File: app-hw-37/src/Main/Main.tsx (hunk 4 above)
Comment (KaterinaPogorelova) [reply app-hw-37/src/Main/Main.tsx:20]: gives TS mistake: Argument of type 'number' is not assignable to parameter of type

File: app-hw-37/src/FullScreenCard/FullScreenCard.tsx (hunk 3 above)
Comment (KaterinaPogorelova) [reply app-hw-37/src/FullScreenCard/FullScreenCard.tsx:21]: gives TS mistake: Argument of type 'null' is not assignable to parameter of type 'Post'

**Pull request merged**